pkg/provisioner/ironic/ironic.go

```diff
@@ -277,24 +277,38 @@ func (p *ironicProvisioner) ValidateManagementAccess(credentialsChanged bool) (r
 			}
 		}
 
-		if p.host.Spec.Image != nil && p.host.Spec.Image.URL != "" {
-			// FIXME(dhellmann): The Stein version of Ironic supports passing
-			// a URL. When we upgrade, we can stop doing this work ourself.
-			checksum, err := p.getImageChecksum()
-			if err != nil {
-				return result, errors.Wrap(err, "failed to retrieve image checksum")
-			}
+		// If there is an image to be provisioned, or an image has
+		// previously been provisioned, include those details. Either
+		// case may mean we are re-adopting a host that was already
+		// known but removed/lost because the pod restarted.
+		//
+		// FIXME(dhellmann): The Stein version of Ironic supports
+		// passing a URL for the checksum. When we upgrade, we can
+		// stop looking up that value ourself.
+		var checksum string
+		var imageURL string
+		switch {
+		case p.host.Spec.Image != nil && p.host.Spec.Image.URL != "":
+			imageURL = p.host.Spec.Image.URL
+			checksum, err = p.getImageChecksum(p.host.Spec.Image.Checksum)
+		case p.host.Status.Provisioning.Image.URL != "":
+			imageURL = p.host.Status.Provisioning.Image.URL
+			checksum, err = p.getImageChecksum(p.host.Status.Provisioning.Image.Checksum)
+		}
```

Comment:
It feels like the priority of these cases should be the other way around - if we had an image provisioned before Ironic restarted then we should adopt using that image. I can't actually think of a case where it should make a difference though.

Reply:
You're probably right, and I imagine I would have hit that working on adoption during provisioning. Can we wait to change the order until I do that work, since that's next on my list?

Reply:
Yeah, I don't think this is a blocker.
(In theory the Status image is empty during provisioning though, so I don't think you will necessarily run into it.)